src/mrack/providers/aws.py

```diff
@@ -69,13 +71,30 @@ async def init(self, image_names, ssh_key, sec_group, instance_tags):
         login_end = datetime.now()
         login_duration = login_end - login_start
         logger.info(f"Login duration {login_duration}")
-        self.images = image_names
+        self.images = images
         self.ssh_key = ssh_key
         self.sec_group = sec_group
         self.instance_tags = instance_tags
 
     async def validate_hosts(self, hosts):
         """Validate that host requirements are well specified."""
+        for req in hosts:
+            req_img = req.get("image")
+            if not req.get("meta_image") and req_img not in self.images.values():
```

Comment:
Let's make sure we understand what's happening here if we are reading this in the future: 
```python
# ...
meta_defined_image = req.get("meta_image")
if not meta_defined_image and req_img not in self.images.values():
    # Check if image name is defined in metadata; if not, check if it's defined in provisioning config
# ...
```

Reply:
In fact yes, but this logic is in transformer.
It loads the `image` and if its `None` it loads the image based on `os` defined in metadata.
This if checks if the `image` was loaded from the metadata and if it was it does not check if the `os` we set in metadata has image set. if not it will raise this error that `os` is not specified in metadata.

After this we just always check image availability.